### PR TITLE
platform: imx8mp: Various cleanups

### PR DESCRIPTION
**Pull request by paulstelian97**
These cleanups are required before merging SDMA driver. They fix various minor issues related to i.MX8MP platform.

* Adjust IRQ_STEER registers for i.MX8MP. The registers are still incorrect for this platform.
* Add extra heap memory. SDMA requests some large allocations (3000+ bytes) so the system heap is increased to accomodate for this.
* Fix DMAC names. i.MX8MP uses SDMA instead of EDMA. Update comments and define names.
* Fix interrupt numbers. SDMA uses a single interrupt number for all channels instead of having separate definitions for separate channels.
* Add SAI FIFO definitions. For some reason these were missing.

Signed-off-by: Paul Olaru <paul.olaru@nxp.com>

@dbaluta There is an additional fix for the DMA channel domain however it depends on the SDMA code itself so I will push it then.

### CODE REVIEW

File: src/platform/imx8m/include/platform/drivers/interrupt.h [hunk 1]
```diff
@@ -105,15 +105,15 @@ int irqstr_get_sof_int(int irqstr_int);
  */
 #define IRQSTR_CHANCTL			0x00
```
Comment (dbaluta) [inline src/platform/imx8m/include/platform/drivers/interrupt.h]: @paulstelian97 can you explain a little bit in the commit message the formula for computing new values. At least why are we replacing n with 5 - n.

**Comment by dbaluta** [comment]
@paulstelian97 mostly looking good. See my one comment about enhancing a little bit the documentation. I know that one can look it up in the Reference Manual, but sometimes is better to have this info in the commit message.

File: src/platform/imx8m/include/platform/drivers/interrupt.h (hunk 1 above)
Comment (dbaluta) [reply src/platform/imx8m/include/platform/drivers/interrupt.h]: even checkpatch.pl cries for this

```
Commit ec0909f72c44 ("drivers: imx: interrupt: Adjust IRQ_STEER registers for i.MX8")
-------------------------------------------------------------------------------------
WARNING: Missing commit description - Add an appropriate one
```

Comment (paulstelian97) [reply src/platform/imx8m/include/platform/drivers/interrupt.h]: I will need to formulate this one better, let me think about how I should comment this.

Comment (paulstelian97) [reply src/platform/imx8m/include/platform/drivers/interrupt.h]: Updated.

**Comment by dbaluta** [comment]
SOFCI TEST

**Pull request merged**